### ISSUE TITLE
Yarn install step not working after Bitrise 11-06-2023 Stack update

### ISSUE THREAD

**Issue by DanielGalarza** [jonoirwinrsa/steps-yarn#27]
### Troubleshooting

- [x] I've searched [discuss.bitrise.io](https://discuss.bitrise.io/) for possible solutions.
- Which version of the step is effected? 1.2.3
- Is the issue reproducible with the latest version? YES 
- Does the issue happen sporadically, or every time? EVERY TIME
- Is the issue reproducible locally by following our [local debug guide](https://devcenter.bitrise.io/troubleshooting/debugging-your-build-on-your-own-machine/)? NOT APPLICABLE

### Useful information

- [FAQ](https://devcenter.bitrise.io/faq/faq-index/)
- [devcenter.bitrise.io](https://devcenter.bitrise.io/)

### Issue description

After the Bitrise stack update from 11-06-2023, the Yarn Install step isn't working for me.
Changelog: https://stacks.bitrise.io/changelogs/linux-docker-android-20.04/#stack-update-v2023-11-06:~:text=Updates%20%23-,Stack%20update,-v2023%2D11%2D06

**Here is my Project's `.tool-versions` file:**
```
ruby 3.2.1
nodejs 18.14.2
yarn 1.22.10
```

### Bitrise info

- Build URL: https://app.bitrise.io/build/04a7aa1a-23e0-4911-bed9-c3c3041eb8d1
- [Bitrise Support](https://devcenter.bitrise.io/troubleshooting/enabling-bitrise-support-user/) enabled: _YES / NO_
- Log: 

  <pre>
      ;[REDACTED]mConfig:
      - workdir: /bitrise/src
      - command: <unset>
      - args: <unset>
      - cache_local_deps: false
      - verbose_log: false
      ;[REDACTED]mYarn is already installed at: /root/.asdf/shims/yarn
      ;[REDACTED]mYarn version:
      ;[REDACTED]m$ yarn "--version"
      No preset version installed for command yarn
      Please install a version by running one of the following:
      
      asdf install nodejs 18.14.2
      
      or add one of the following versions in your config file at /bitrise/src/.tool-versions
      
      nodejs 16
      nodejs 16.20
      nodejs 16.20.2
      [REDACTED];[REDACTED]mInstall dependencies: yarn version command failed: exit status [REDACTED]26
  </pre>

### Steps to reproduce

Run Yarn Install step any multistep workflow:
<img width="385" alt="Screenshot 2023-11-09 at 4 08 52 PM" src="https://github.com/bitrise-steplib/steps-yarn/assets/8992942/37c15e16-f00d-4ec0-9e47-18d8a2e65d98">

**Comment by DanielGalarza** [comment]
It seems like it wants me to downgrade my Node version to 16.x.x, but this wasn't a requirement before.  If I have to install  nodejs 18.14.2, what's the best way to do that?

Adding a script step with this command didn't work:
```
asdf install nodejs 18.14.2
```

**Comment by DanielGalarza** [comment]
Might be able to close this if #26 is still active. That seems to be a similar issue, although that was earlier this year. Folks are commenting on that one (#26) for this exact issue.

**Comment by ofalvai** [comment]
@DanielGalarza the ASDF error message is a bit misleading about installing Node itself, what you need to do is `npm install -g yarn`. Yarn is installed as a global NPM package for the default Node version, it means that if you switch versions via `.tool-versions`, the Yarn binary is going to be missing.

Apologies for this weird edge case, we are looking into a more consistent Yarn experience (we are investigating enabling [corepack](https://nodejs.org/api/corepack.html), for example)

**Comment by DanielGalarza** [comment]
@ofalvai Do you recommend running `npm install -g yarn` with the `Run npm command` bitrise step, OR as a generic script step?

**Comment by DanielGalarza** [comment]
Okay, looks like the recommendation worked. I used these steps in my Bitrise workflow to get Yarn to work again:
1. Added Node Version Manager (used to set Node to the version I have in my .tools-version file, nodejs 18)
2. Added Run NPM command step that runs `npm install -g yarn`

My Yarn step is unchanged, but will run after the steps above

<img width="389" alt="Screenshot 2023-11-13 at 10 29 16 AM" src="https://github.com/bitrise-steplib/steps-yarn/assets/8992942/5417a6f5-f942-4b07-9327-2b0ae7520441">
